lib/kafka/transaction_manager.rb

```diff
@@ -286,8 +286,8 @@ def retry_error_handler
       if @retry_counter > @max_retries
         raise e
       end
-      @logger.info("#{e.class.name}, sleeping #{@retry_backoff} , retrying...")
-      sleep @retry_backoff
+      @logger.info("#{e.class.name}, sleeping #{(@retry_counter * @retry_backoff)} seconds, retrying...")
+      sleep(@retry_counter * @retry_backoff)
```

Comment:
we start with 20 milliseconds as in Java client, then increase amount each time with 20 milliseconds